src/routes/rosters/+page.svelte

```diff
@@ -1,63 +1,97 @@
 <script lang="ts">
-    let showAddModal = false
-    let teams: number[] = [1, 2, 3, 4, 5]
-
+    import {
+      type team,
+      teams,
+      addTeam
+    } from './rosters.svelte';
+  
+    let showAddModal = false;
     const openAddModal = () => (showAddModal = true);
     const closeAddModal = () => (showAddModal = false);
-</script>
-
 
-<title>Rosters</title>
-
-<div>
-    <section class= "rosters-dash">
-        <h1>Rosters</h1>
-        <h3>A List of All Teams on the Website</h3>
+    // Local state for new team data
+    let newTeamName: string = "";
+    let newTeamHometown: string = "";
+    let newCoachName: string = "";
+  
+    // Handler for form submission to add a team
+    function handleAddTeam(event: Event) {
+      event.preventDefault();
+      // Create a new team object; note that players is empty initially.
+      const newTeam: team = {
+        name: newTeamName,
+        hometown: newTeamHometown,
+        coach: newCoachName,
+        players: []
+      };
+  
+      // Add the new team to the teams state
+      addTeam(newTeam);
+  
+      // Reset form fields
+      newTeamName = "";
+      newTeamHometown = "";
+      newCoachName = "";
+  
+      // Close the modal
+      closeAddModal();
+    }
```

Comment:
The problem with using bind here, as you've already addressed, is that the binding goes both ways so you have to reset the form field values every time you submit. Forms automatically do this unless you use bind, so you shouldn't use bind and instead just read out the values directly from the form submission event.
Probably something like this:
```suggestion
   // Handler for form submission to add a team
    function handleAddTeam(event) {
      event.preventDefault();

      let formData = new FormData(event.target);
      
      // Create a new team object; note that players is empty initially.
      const newTeam: team = {
        name: formData.get("team-name").toString(),
        hometown: formData.get("hometown").toString(),
        coach: formData.get("coach-name").toString(),
        players: []
      };
  
      // Add the new team to the teams state
      addTeam(newTeam);
  
      // Close the modal
      closeAddModal();
    }
```